composition/cache_loader.go

```diff
@@ -30,7 +33,18 @@ func (loader *CachingContentLoader) Load(fd *FetchDefinition) (Content, error) {
 	c, err := loader.load(fd)
 	if err == nil {
 		if fd.IsCachable(c.HttpStatusCode(), c.HttpHeader()) {
-			loader.cache.Set(hash, fd.URL, c.MemorySize(), c)
+			if c.Reader() != nil {
+				var streamBytes []byte
+				streamBytes, err = ioutil.ReadAll(c.Reader())
```

Comment:
The error is nor handled:
If an error occurs here, while reading, than we would cache the broken response.